hyperspy/component.py

```diff
@@ -1216,3 +1216,40 @@ def print_current_values(self, only_free=False, fancy=True):
         else:
             display_pretty(current_component_values(
                 self, only_free=only_free))
+
+
+def _get_scaling_factor(signal, axis, parameter):
+    """
+    Convenience function to get the scaling factor required to take into
+    account binned and/or non uniform axes.
+
+    Parameters
+    ----------
+    signal : BaseSignal
+    axis : BaseDataAxis
+    parameter : float or numpy array
+        The axis value at which scaling factor is evaluated (ignored if the axis
+        is uniform)
+
+    Returns
+    -------
+    scaling_factor
+
+    """
+
+    if is_binned(signal):
+    # in v2 replace by
+    #if axis.is_binned:
+        if axis.is_uniform:
+            scaling_factor = axis.scale
+        else:
+            # if isinstance(parameter, (int, float)):
```

Comment:
Now that `value2index` takes array as input, this can be clean up.

Reply:
Sure, I just forgot